MaxText/train.py

```diff
@@ -226,22 +231,16 @@ def train_loop(config, state=None):
   mesh = Mesh(devices_array, config.mesh_axes)
 
   # Model and Optimizer definition
-  model = Transformer(config, mesh)
+  model = max_utils.get_model(config, mesh)
```

Comment:
lets separate out maxtext_utils and max_utils.

This is maxtext_utils, not max utils.

Reply:
I was silly to duplicate the initializer of Transfomer in a utils file, one should just call the Transformer.

It is probably still a good idea to split max_utils into train.py specific maxtext_utils and shared max_utils, especially since max_utils is quite large now. This can be done in a followup as this is already a large PR